subsys/bluetooth/controller/hci_driver.c

```diff
@@ -250,8 +250,18 @@ void sdc_assertion_handler(const char *const file, const uint32_t line)
 #else /* !IS_ENABLED(CONFIG_BT_CTLR_ASSERT_HANDLER) */
 void sdc_assertion_handler(const char *const file, const uint32_t line)
 {
+#if defined(CONFIG_ASSERT) && defined(CONFIG_ASSERT_VERBOSE) && !defined(CONFIG_ASSERT_NO_MSG_INFO)
+	__ASSERT(false, "SoftDevice Controller ASSERT: %s, %d\n", file, line);
+#elif defined(CONFIG_LOG)
 	LOG_ERR("SoftDevice Controller ASSERT: %s, %d", file, line);
 	k_oops();
+#elif defined(CONFIG_PRINTK)
+	printk("SoftDevice Controller ASSERT: %s, %d\n", file, line);
```

Comment:
Would it be an idea to add "\<err\> " to the start of the lines for printk, or will that come through the string?

Reply:
The `k_oops()` below will print out a fatal error message

Reply:
I think it would be nicer to let the application run, maybe it can recover from this situation by re-initializing `sdc`?

Reply:
It may be hard to recover from such unexpected situations. We trigger these when the HW or stack is in an unexpected state. Rebooting the application is currently the most fool-proof solution of recovery